src/main/java/org/wso2/carbon/inbound/vfs/Utils.java

```diff
@@ -0,0 +1,715 @@
+        if (outDiskShareAccessMasks.isEmpty()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Set the access mask to default MAXIMUM_ALLOWED since the access mask is not defined or the defined values are not valid.");
+            }
+            outDiskShareAccessMasks.add(DISK_SHARE_ACCESS_MASK_MAX_ALLOWED);
+        }
+
+        return outDiskShareAccessMasks;
+    }
+
+    public static boolean isFailRecord(FileSystemManager fsManager, FileObject fo, FileSystemOptions fso) {
+        try {
+            String fullPath = fo.getName().getURI();
+            String queryParams = "";
+            int pos = fullPath.indexOf('?');
+            if (pos > -1) {
+                queryParams = fullPath.substring(pos);
+                fullPath = fullPath.substring(0, pos);
+            }
+            FileObject failObject = fsManager.resolveFile(fullPath + FAIL_FILE_SUFFIX + queryParams, fso);
+            if (failObject.exists()) {
+                return true;
+            }
+        } catch (FileSystemException e) {
+            log.error("Couldn't release the fail for the file : " + maskURLPassword(fo.getName().getURI()));
+        }
+        return false;
+    }
+
+    public static boolean isFailedRecordInFailedList(FileObject fileObject, VFSConfig entry) {
+        String failedFile = entry.getFailedRecordFileDestination() +
+                entry.getFailedRecordFileName();
+        File file = new File(failedFile);
+        if (file.exists()) {
+            try {
+                List list = FileUtils.readLines(file);
+                for (Object aList : list) {
+                    String str = (String) aList;
+                    StringTokenizer st = new StringTokenizer(str,
+                            org.wso2.carbon.inbound.vfs.VFSConstants.FAILED_RECORD_DELIMITER);
+                    String fileName = st.nextToken();
+                    if (fileName != null &&
+                            fileName.equals(fileObject.getName().getBaseName())) {
+                        return true;
+                    }
+                }
+            } catch (IOException e) {
+                log.fatal("Error while reading the file '"
+                        + Utils.maskURLPassword(failedFile) + "'", e);
+            }
+        }
+        return false;
+    }
+
+    public static String extractPath(String uri) {
+        int queryIndex = uri.indexOf('?');
+        if (queryIndex != -1) {
+            return uri.substring(0, queryIndex);
+        }
+        return uri;
+    }
+
+    /**
+     * Release a file item lock acquired either by the VFS listener or a sender
+     *
+     * @param fsManager which is used to resolve the processed file
+     * @param fo        representing the processed file
+     * @param fso       represents file system options used when resolving file from file system manager.
+     */
+    public static void releaseLock(FileSystemManager fsManager, FileObject fo, FileSystemOptions fso) {
```

Comment:
Shall we move this to `LockManager`?

Reply:
This utility has been used in other places where we are not using the lock instance; that is why I moved it to utils.

Reply:
Moved